src/bhattacharyya.jl

```diff
@@ -50,6 +50,34 @@ end
     return sqab, asum, bsum
 end
 
+@inline function _bhattacharyya_coeff(a::SparseVectorUnion{<:Number}, b::SparseVectorUnion{<:Number})
+    anzind = nonzeroinds(a)
+    bnzind = nonzeroinds(b)
+    anzval = nonzeros(a)
+    bnzval = nonzeros(b)
+    ma = nnz(a)
+    mb = nnz(b)
+
+    ia = 1; ib = 1
+    eval_op = (a, b) -> sqrt(a * b)
+    s = zero(eval_op(zero(eltype(a)), zero(eltype(b))))
+    @inbounds while ia <= ma && ib <= mb
+        ja = anzind[ia]
+        jb = bnzind[ib]
+        if ja == jb
+            v = eval_op(anzval[ia], bnzval[ib])
+            s = v + s
+            ia += 1; ib += 1
+        elseif ja < jb
+            ia += 1
+        else
+            ib += 1
+        end
+    end
+    # efficient method for sum for SparseVectorView is missing
+    return s, sum(anzval), sum(bnzval)
```

Comment:
Don't you want to accumulate `anzval` and `bnzval` in the loop above, similar to the dense version?

Reply:
I thought about it:
* Accumulating in loops takes some thought and is error prone. For example, I would need to add two more loops to accumulate properly (compare to the last two loops in the similar code for `UnionMetrics`.
* The accumulation happens inside conditionals. On the other hand, with `sum` an entire packed array is summed. This allows optimizations, such as simd (But, I don't know if `sum` employs any of these for `Vector{Float64}` since it uses pairwise summation.)
* I did not benchmark the different possibilities.

Reply:
I see, I think your second point alone is good enough.